src/app/tabs/newAdmin/resolver/operations.js

```diff
@@ -199,3 +206,91 @@ export const setContent = (contentType, resolverAddress, domain, value) => (disp
     dispatch(setContentBytes(resolverAddress, domain, value));
   }
 };
+
+/**
+ * Set the resolver to the Definitive Resolver and Migrate Users Addresses
+ * @param {string} domain domain to be migrated
+ * @param {array} chainAddresses array of all the chainAddresses from the reducer
+ * @param {bool} understandWarning bool that the user knows some addresses are invalid
+ */
+export const setDomainResolverAndMigrate = (
+  domain, chainAddresses, understandWarning,
+) => async (dispatch) => {
+  dispatch(requestMigrateAddresses());
+  const accounts = await window.ethereum.enable();
+  const currentAddress = accounts[0];
+  const hash = namehash(domain);
+
+  const definitiveResolver = new web3.eth.Contract(
+    definitiveResolverAbi, definitiveResolverAddress, { gasPrice: defaultGasPrice },
+  );
+
+  // convert JSON object into array and filter out empty items:
+  const nonEmpties = Object.entries(chainAddresses)
+    .filter(item => item[1].address !== '' && item[1].address !== EMPTY_ADDRESS);
+
+  // loop through nonEmpties and to get decoded version of the address, if valid, create
+  // the contract method and add it to the multiCallMethods array.
+  const multiCallMethods = [];
+  let decodeError = false;
+  nonEmpties.forEach((item) => {
```

Comment:
We are iterating twice this array
- if empty -> do nothing
- if !empty + valid -> add
- if !empty + !valid -> notify